docs(block-node-overview): fix swapped Getting Started links and improved documentation accuracy. (#2759)
hiero-ledger/hiero-block-node: docs/block-node/block-node-overview.md

Reviewer: This should possibly link to the [HW spec doc](https://github.com/hiero-ledger/hiero-block-node/blob/main/docs/block-node/operations/block-node-hardware-specifications.md), not the deployment guide.

# Block Node Overview

Block Nodes represent a new class of nodes in a Hiero network designed to increase decentralization and network data distributions. They enable operators to assume responsibility for long-term block and state storage while supporting the security and performance characteristics of a Hiero network.

This overview document provides operators with the essential concepts needed to understand Block Node roles and responsibilities before diving into deployment.

## What is a Block Node?

A Block Node is a special kind of server that keeps a complete, trustworthy copy of what is happening on a Hiero network.

> It receives a stream of already-agreed blocks from Consensus Nodes, checks that each block is valid, and then stores those blocks and serves state snapshots so both blocks and state can be queried later.

Reviewer 2: ```suggestion
It receives a stream of already-agreed blocks from Consensus Nodes, checks that each block is valid, and then stores those blocks and serves single block and block streams via API queries. In the future Block Nodes will also maintain a valid copy of the current network state and will serve state related queries via new APIs.
```

> It receives a stream of already-agreed blocks from Consensus Nodes, checks that each block is valid, and then stores those blocks and serves single block and block streams via API queries. In the future Block Nodes will also maintain a valid copy of the current network state and will serve state related queries via new APIs.

Reviewer 3: Latest update made this a bit awkward.
Possible adjustment (not required)

```suggestion
It receives a stream of already-agreed blocks from Consensus Nodes, delivers the block stream via the subscribe API, checks that each block is valid, stores valid blocks, and serves single blocks via API query. In the future Block Nodes will also maintain an accurate copy of the current network state and will serve state related queries via new APIs.
```

|--------------------------------|---------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|
| Primary role                   | Reach consensus and update canonical state.                                                       | Ingest, verify, store, and serve blocks, and state.                                                             | Provide value‑added access to historical data and analytics.                                                                       |
| Produces blocks                | Yes – produces finalized block streams per **([HIP-1056](https://hips.hedera.com/hip/hip-1056))** | No – consumes and verifies blocks from Consensus Nodes or upstream Block Nodes.                                 | No – will consume data from Block Nodes once the cutover lands; today Mirror Nodes still download record files from cloud storage. |

> | Maintains full consensus state | Yes – authoritative state, optimized for consensus.                                               | Manages an active copy of network state locally, updated with `State Changes`; supports reconnect flows.        | No – maintains data in an indexed form as needed for queries and analytics.                                                        |

Reviewer 2: Today BN doesn't yet maintain state but will in the near/medium term
```suggestion
| Maintains full consensus state | Yes – authoritative state, optimized for consensus.                                               | (planned) Manages an active copy of network state locally, updated with `State Changes`; supports reconnect flows.        | No – maintains data in an indexed form as needed for queries and analytics.                                                        |
```

## Key Terms

Before diving deeper, familiarize yourself with these core concepts:

- **Block Stream ([HIP-1056](https://hips.hedera.com/hip/hip-1056))** - A continuous, ordered feed of finalized block data produced by Consensus Nodes. Each block consists of a stream of individual items containing transactions, state changes, events, EVM trace data, and cryptographic proofs.
  Block streams are delivered via gRPC in Protocol Buffer format at a rate determined by network configuration and network usage. A typical public network might complete 1 block per second, up to several blocks per second.

- **State Snapshot** - A point-in-time capture of the complete network state (accounts, balances, smart contract storage, etc.) at a specific block height. State snapshots enable fast synchronization and recovery without replaying every transaction from genesis.

> - **Aggregated Signatures** - Cryptographic signatures from multiple Consensus Nodes combined into a single compact signature. Verification uses the network's Ledger ID — the hash of the genesis TSS Address Book — to confirm a block was finalized by network consensus. For post-genesis blocks, **WRAPS (Weighted Roster Attestation Proof System)** proofs attest that the active roster is a valid descendant of the genesis roster, ensuring verification remains trustworthy as the roster evolves over time. Defined in [HIP-1200](https://hips.hedera.com/hip/hip-1200); the trusted-setup ceremony that produces the WRAPS public parameters is specified in [HIP-1398](https://github.com/hiero-ledger/hiero-improvement-proposals/pull/1398) (draft).

Reviewer 2: @jsync-swirlds to confirm we're using roster here and moving away from address book
```suggestion
- **Aggregated Signatures** - Cryptographic signatures from multiple Consensus Nodes combined into a single compact signature. Verification uses the network's Ledger ID — the hash of the genesis TSS Roster — to confirm a block was finalized by network consensus. For post-genesis blocks, **WRAPS (Weighted Roster Attestation Proof System)** proofs attest that the active roster is a valid descendant of the genesis roster, ensuring verification remains trustworthy as the roster evolves over time. Defined in [HIP-1200](https://hips.hedera.com/hip/hip-1200); the trusted-setup ceremony that produces the WRAPS public parameters is specified in [HIP-1398](https://github.com/hiero-ledger/hiero-improvement-proposals/pull/1398) (draft).
```

Author: changed it to Roster as @jsync-swirlds used this term for another document too.

Reviewer 3: Confirmed.  TSS Roster is based on the Node Store, but is not the same as the "Address Book" (which is mostly obsolete at this point).

> - **Aggregated Signatures** - Cryptographic signatures from multiple Consensus Nodes combined into a single compact signature. Verification uses the network's Ledger ID — the hash of the genesis TSS Roster — to confirm a block was finalized by network consensus. For post-genesis blocks, **WRAPS (Weighted Roster Attestation Proof System)** proofs attest that the active roster is a valid descendant of the genesis roster, ensuring verification remains trustworthy as the roster evolves over time. Defined in [HIP-1200](https://hips.hedera.com/hip/hip-1200); the trusted-setup ceremony that produces the WRAPS public parameters is specified in [HIP-1398](https://github.com/hiero-ledger/hiero-improvement-proposals/pull/1398) (draft).

Reviewer 3: I believe HIP-1398 is no longer draft.

```suggestion
- **Aggregated Signatures** - Cryptographic signatures from multiple Consensus Nodes combined into a single compact signature. Verification uses the network's Ledger ID — the hash of the genesis TSS Roster — to confirm a block was finalized by network consensus. For post-genesis blocks, **WRAPS (Weighted Roster Attestation Proof System)** proofs attest that the active roster is a valid descendant of the genesis roster, ensuring verification remains trustworthy as the roster evolves over time. Defined in [HIP-1200](https://hips.hedera.com/hip/hip-1200); the trusted-setup ceremony that produced the WRAPS public parameters is specified in [HIP-1398](https://github.com/hiero-ledger/hiero-improvement-proposals/pull/1398).
```

# Block Node Overview

Block Nodes represent a new class of nodes in a Hiero network designed to increase decentralization and network data distributions. They enable operators to assume responsibility for long-term block and state storage while supporting the security and performance characteristics of a Hiero network.

This overview document provides operators with the essential concepts needed to understand Block Node roles and responsibilities before diving into deployment.

## What is a Block Node?

A Block Node is a special kind of server that keeps a complete, trustworthy copy of what is happening on a Hiero network.

> It receives a stream of already-agreed blocks from Consensus Nodes, checks that each block is valid, and then stores those blocks and serves verifiable state snapshots so both blocks and state can be queried later.

Reviewer 3: This was a bullet somewhat on purpose, to _not_ have it separated by a paragraph break.
Ideally we'd have line breaks, but the Hiero spotless formatting makes that difficult.

|--------------------------------|---------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|
| Primary role                   | Reach consensus and update canonical state.                                                       | Ingest, verify, store, and serve blocks, and state.                                                             | Provide value‑added access to historical data and analytics.                                                                       |
| Produces blocks                | Yes – produces finalized block streams per **([HIP-1056](https://hips.hedera.com/hip/hip-1056))** | No – consumes and verifies blocks from Consensus Nodes or upstream Block Nodes.                                 | No – will consume data from Block Nodes once the cutover lands; today Mirror Nodes still download record files from cloud storage. |

> | Maintains full consensus state | Yes – authoritative state, optimized for consensus.                                               | Stores and serves state snapshots produced by Consensus Nodes; supports reconnect flows.                        | No – maintains data in an indexed form as needed for queries and analytics.                                                        |

Reviewer 3: Block node does not receive snapshots from consensus nodes:

```suggestion
| Maintains full consensus state | Yes – authoritative state, optimized for consensus.                                               | Manages an active copy of network state locally, updated with `State Changes`; supports reconnect flows.                        | No – maintains data in an indexed form as needed for queries and analytics.                                                        |
```

Block node _also_ produces state snapshots (CN will no longer upload state snapshots).

- **Reconnect Services** - APIs and data streams that will help Consensus Nodes catch up to the current network state after downtime or network partitions by providing recent blocks and state snapshots. Service interfaces are defined; this capability is currently in active development for a future release.

- **Tier 1 Block Node** – A Block Node that receives block streams directly from Consensus Nodes and typically provides reconnect/state snapshot services back to the network.

- **Tier 2 Block Node** – A Block Node that receives block streams from one or more upstream Block Nodes (Tier 1 or another Tier 2) and is permissionless to operate.

## Role in the Hiero Network

> Block Nodes act as the trusted historians and data providers for a Hiero network. They receive block streams from Consensus Nodes, verify that each block and its data are correct, and durably store the blocks together with the verifiable state snapshots produced by the network.

Reviewer 3: The other version was correct, this is not.
The network does not produce state snapshots stored by block nodes.
Block nodes maintain a single active copy of network state (as a MerkleDB, the same as consensus nodes), update it with values from the `StateChanges` in blocks, and _produce_ snapshots.

The consensus nodes no longer upload snapshots once block nodes have state management.

Also, state snapshots are not, and have never been, verifiable.

- Durable storage of blocks, and consensus state.
- Real-time and historical data streaming to downstream clients.
- Random-access retrieval of blocks and state at specific block heights.

> - State snapshot storage and serving, plus reconnect services (currently in active development) to help Consensus Nodes catch up or recover.

Reviewer 3: Again, block nodes _create_ state snapshots, they don't "store and serve" snapshots created elsewhere.
Also, reconnect is a future development, it is _not_ currently in active development.

```suggestion
- State snapshot creation, plus reconnect services (not currently in active development) to help Consensus Nodes catch up or recover.
```

## Example Use Cases

- Running an API server for wallets, explorers, or dApps to access verified blockchain data.
- Operating analytics or aggregation pipelines using live and historical block streams.
- Providing specialized compliance, archival, or network recovery support for Hiero services.

## Block Node Tiers

A Block Node's *tier* describes where it gets its block stream from. The same core software runs at every tier; the differences are operational.

| **Tier** |                         **Description**                          |        **Typical Operators**        |               **Key Focus**               |
|----------|------------------------------------------------------------------|-------------------------------------|-------------------------------------------|

> | Tier 1   | Receive streams directly from Consensus Nodes; high reliability. | Governing Council, trusted entities | Verification, reconnect, state snapshots. |

Reviewer 3: ```suggestion
| Tier 1   | Receive streams directly from Consensus Nodes; high reliability. | Governing Council, trusted entities | Verification, reconnect, state management. |
```

| Tier 2   | Receive streams from Tier 1 or another Tier 2; permissionless.   | Community, enterprises              | Streaming, proofs, geographic redundancy. |

> Beyond tiers, operators can deploy a Block Node in different *types* — for example *Full Node*, *Rolling-History*, *Light Node*, *Private-Cloud*, *Archive Server*, or *Community Node* — by combining different sets of plugins. *Rolling-History* provides a Partial History service, retaining only recent blocks rather than full history. *Archive Server* provides cold storage and on-demand retrieval without live streaming. See [Block-Node-Types.md](../Block-Node-Types.md) for the full taxonomy.

Reviewer 3: ```suggestion
Beyond tiers, operators can deploy a Block Node in different *types* — for example *Full Node*, *Rolling-History*, *Light Node*, *Private-Cloud*, *Archive Server*, or *Community Node* — by combining different sets of plugins. *Rolling-History* provides a Partial History service, retaining only recent blocks rather than full history. *Archive Server* provides cold storage without live streaming. See [Block-Node-Types.md](../Block-Node-Types.md) for the full taxonomy.
```

## High-Level Architecture

Block Nodes follow a modular design, receiving **block streams**, verifying integrity using **aggregated signatures** and cryptographic mechanisms defined in related HIPs ([HIP-1200](https://hips.hedera.com/hip/hip-1200) and [HIP-1056](https://hips.hedera.com/hip/hip-1056)), and persisting blocks and state to local disk and (optionally) S3-compatible archival storage.
They may provide four common functions:

- **Block Stream ingestion and verification** - Receives block streams from Consensus Nodes and verifies their integrity using aggregated signatures and Merkle proofs.

> - **State snapshot management** - Stores the verifiable state snapshots produced by Consensus Nodes and serves them through dedicated APIs to support reconnect and recovery flows. Live, fully replicated consensus state on Block Nodes is a roadmap goal, not a current capability.

Reviewer 3: This line is completely incorrect.
The previous text is correct.
Block nodes _do not_ manage snapshots produced by consensus nodes, and will not do so.
Block Nodes _will_ maintain and manage a "live" copy of state and _generate_ snapshots. Management of live state is a _prerequisite_ to reconnect support, and neither is currently implemented.
